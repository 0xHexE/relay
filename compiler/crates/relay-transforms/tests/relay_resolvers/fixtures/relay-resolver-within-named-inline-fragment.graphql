fragment Foo_user on Node {
  ... on User @alias(as: "named_fragment") {
    # Ensure the field_path we derive inlcudes "named_fragment"
    pop_star_name
  }
}

fragment PopStarNameResolverFragment_name on User {
  parents {
    lastName
  }
}

# %extensions%

extend type User {
  pop_star_name: String @relay_resolver(fragment_name: "PopStarNameResolverFragment_name", import_path: "PopStarNameResolver")
}
