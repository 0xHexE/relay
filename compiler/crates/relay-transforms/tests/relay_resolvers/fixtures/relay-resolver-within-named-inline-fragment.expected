==================================== INPUT ====================================
fragment Foo_user on Node {
  ... on User @alias(as: "named_fragment") {
    # Ensure the field_path we derive inlcudes "named_fragment"
    pop_star_name
  }
}

fragment PopStarNameResolverFragment_name on User {
  parents {
    lastName
  }
}

# %extensions%

extend type User {
  pop_star_name: String @relay_resolver(fragment_name: "PopStarNameResolverFragment_name", import_path: "PopStarNameResolver")
}
==================================== OUTPUT ===================================
fragment Foo_user on Node {
  ... on User @__FragmentAliasMetadata
  # FragmentAliasMetadata {
  #     alias: WithLocation {
  #         location: relay-resolver-within-named-inline-fragment.graphql:49:51,
  #         item: "named_fragment",
  #     },
  #     type_condition: Some(
  #         Object(68),
  #     ),
  # }
   {
    ...PopStarNameResolverFragment_name @__RelayResolverSpreadMetadata
    # RelayResolverSpreadMetadata {
    #     field_parent_type: "User",
    #     import_path: "PopStarNameResolver",
    #     field_name: "pop_star_name",
    #     field_alias: None,
    #     field_path: "named_fragment.pop_star_name",
    #     field_arguments: [],
    #     live: false,
    # }
    
  }
}

fragment PopStarNameResolverFragment_name on User {
  parents {
    lastName
  }
}

# Implicit Dependencies:
#
# Foo_user --> {PopStarNameResolverFragment_name}
